includes/plugins/wc-memberships/class-memberships.php

```diff
@@ -774,15 +773,18 @@ public static function render_js() {
 				ras.on( 'reader', function( ev ) {
 					if ( ev.detail.authenticated && ! window?.newspackReaderActivation?.getPendingCheckout() ) {
 						if ( ras.overlays.get().length ) {
+							// When an overlay is added or removed,
+							// check if there are none – this means an overlay was removed and there are none left.
+							// In this case, reload the window.
 							ras.on( 'overlay', function( ev ) {
-								if ( ! ev.detail.overlays.length && ! window?.newspackReaderActivation?.openNewslettersSignupModal ) {
+								if ( ! ev.detail.overlays.length ) {
 									window.location.reload();
 								}
 							} );
 						} else {
 							setTimeout( function() {
 								window.location.reload();
-							}, 2000 );
+							}, 2000 ); // HACK: 2s delay to allow the purchase to be processed.
```

Comment:
This is not a hack to allow the purchase to be processed. It's for UX, so the "Thank you" message at the end of the transaction is not just flashed to the reader before the page refresh.

Reply:
That makes more sense, I've just rephrased the original function comment:

> reload the page after 2 seconds in case RAS detects a new reader. This allows the membership purchase to unlock the content.

I'll test and update the comment.

Reply:
I'm missing something here. At the end of the modal checkout transaction there's a success state with a button (which [is rendered unconditionally](https://github.com/Automattic/newspack-blocks/blob/trunk/src/modal-checkout/templates/thankyou.php#L69-L71)):

<img width="456" alt="image" src="https://github.com/user-attachments/assets/6b0ba4d4-9855-4db6-8dbb-6c9b14d0caab" />

Which steps does the user have to take to arrive at the condition with the `setTimeout`?

Reply:
I've removed this code for completeness sake, but once I have the reproduction steps for the case that this was supposed to handle, I'll work on accommodating it.

Reply:
You might be able to reproduce it by using the "Sign In" to authenticate to a member account while accessing a locked article.

Reply:
That's the scenario I'm testing: 

1. visit a reg-walled article
2. use the "Sign In" (or manually placed button) to authenticate

But on `trunk`, the `if ( ras.overlays.get().length )` is always met, so the else-branch (with the 2s delay) is never executed.

Can you test the scenario you have on mind on this branch?

Reply:
Ah, I see. In that case, I can only imagine it being useful for other opened tabs, which is a very edge case that I don't think justifies the additional logic.